src/Text/Pandoc/Highlighting.hs

```diff
@@ -52,12 +52,12 @@ highlightingStyles =
    ("breezedark", breezeDark),
    ("haddock", haddock)]
 
-languages :: [T.Text]
-languages = [T.toLower (sName s) | s <- M.elems defaultSyntaxMap]
+languages :: SyntaxMap -> [T.Text]
+languages syntaxmap = [T.toLower (sName s) | s <- M.elems syntaxmap]
 
-languagesByExtension :: T.Text -> [T.Text]
-languagesByExtension ext =
-  [T.toLower (sName s) | s <- syntaxesByExtension defaultSyntaxMap (T.unpack ext)]
+languagesByExtension :: SyntaxMap -> T.Text -> [T.Text]
+languagesByExtension syntaxmap ext =
+  [T.toLower (sName s) | s <- syntaxesByExtension syntaxmap (T.unpack ext)]
 
```

Comment:
Note that this would be a breaking API change.  (Not a reason against, necessarily, but should be noted very clearly in the commit log.)

Reply:
Should I add something like `+ Text.Pandoc.Highlighting: languages and languagesByExtension now take an additional SyntaxMap argument. Get it from writerOpts or use Skylighting’s defaultSyntaxMap. [API change]`.

Or I can create new functions and make the old ones call them with `defaultSyntaxMap`.